File: benchmark/output/lua-love/openai/pendulum_simulation.lua
local angle = 1.0
local angular_vel = 0.0
local length = 250.0
local gravity = 9.8

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Simple Pendulum")
end

function love.update(dt)
    local angular_acc = (0.0 - gravity / length) * math.sin(angle)
    angular_vel = angular_vel + angular_acc * dt
    angular_vel = angular_vel * 0.999
    angle = angle + angular_vel * dt
end

function love.draw()
    local bob_x = 400.0 + math.sin(angle) * length
    local bob_y = 100.0 + math.cos(angle) * length

    love.graphics.setColor(1, 1, 1)
    love.graphics.circle("fill", 400.0, 100.0, 5.0)

    for i = 0, 19 do
        local t = i / 20.0
        local rx = 400.0 + math.sin(angle) * length * t
        local ry = 100.0 + math.cos(angle) * length * t
        love.graphics.rectangle("fill", rx, ry, 2.0, 2.0)
    end

    love.graphics.circle("fill", bob_x, bob_y, 15.0)

    love.graphics.print("Angle: " .. string.format("%.2f", angle), 10.0, 10.0)
    love.graphics.print("Click to push", 10.0, 30.0)
end

function love.mousepressed(mx, my, button)
    if button == 1 then
        angular_vel = angular_vel + 0.5
    end
end

File: vibe-lang/benchmark/output/python-pygame/openai/pendulum_simulation.py
import pygame
import sys
import math

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Simple Pendulum")
clock = pygame.time.Clock()

# Pendulum state variables
angle = 1.0
angular_vel = 0.0
length = 250.0
gravity = 9.8

def update(dt):
    global angle, angular_vel
    angular_acc = (0.0 - gravity / length) * math.sin(angle)
    angular_vel += angular_acc * dt
    angular_vel *= 0.999  # Damping
    angle += angular_vel * dt

def draw():
    screen.fill((0, 0, 0))
    
    # Compute bob position
    bob_x = 400.0 + math.sin(angle) * length
    bob_y = 100.0 + math.cos(angle) * length
    
    # Draw pivot
    pygame.draw.circle(screen, (255, 255, 255), (400, 100), 5)
    
    # Draw rod
    for i in range(0, 20):
        t = float(i) / 20.0
        rx = 400.0 + math.sin(angle) * length * t
        ry = 100.0 + math.cos(angle) * length * t
        pygame.draw.rect(screen, (255, 255, 255), (rx, ry, 2, 2))
    
    # Draw bob
    pygame.draw.circle(screen, (255, 255, 255), (int(bob_x), int(bob_y)), 15)
    
    # Display text
    font = pygame.font.Font(None, 36)
    angle_text = font.render(f"Angle: {angle:.2f}", True, (255, 255, 255))
    screen.blit(angle_text, (10, 10))
    click_text = font.render("Click to push", True, (255, 255, 255))
    screen.blit(click_text, (10, 30))

def mousepressed(mx, my, button):
    global angular_vel
    angular_vel += 0.5

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.MOUSEBUTTONDOWN:
            mousepressed(*pygame.mouse.get_pos(), event.button)

    update(dt)
    draw()
    pygame.display.flip()

pygame.quit()
sys.exit()